### PR TITLE
add generic ModulesRC easyblock

### PR DESCRIPTION
**Pull request by boegel**
This is a new generic easyblock that generates a `.modulerc` file to define a module version alias rather than installing an actual module file (cfr. suggestion by @mboisson in https://github.com/easybuilders/easybuild-easyconfigs/pull/5203#issuecomment-415827214).

This can be used to for example install a `Java/1.8` wrapper (see https://github.com/easybuilders/easybuild-easyconfigs/pull/6712) for another `Java` module, without causing trouble with for example Lmod's module collection concept...

edit: now requires ~~https://github.com/easybuilders/easybuild-framework/pull/2571~~ and ~~https://github.com/easybuilders/easybuild-framework/pull/2575~~

### CODE REVIEW

**Comment by boegel** [comment]
The tests only fail with EnvironmentModulesC 3.2.10, looks like there was a weird bug with `module-version` there, see https://sourceforge.net/p/modules/mailman/message/31144974 + https://sourceforge.net/p/modules/mailman/message/33399425...

@xdelaruelle Any ideas what's going on here? In this case, there's only a singe `module-version` statement in the `.modulerc` file, like this:

```
#%Module
module-version Java/1.8.0_181 1.8
```

Then how can it trigger a `Duplicate version symbol` error?!
```

**Comment by boegel** [comment]
OK, the problem was probably that there was an actual module file `foo/1.3.2` together with a module alias for `foo` defined via `module-version` in `.modulerc` that was also using `1.3.2` as version, which results in a clash...

That should be fixed now by user different versions for the `ModulesAlias` test.

**Comment by xdelaruelle** [comment]
@boegel It seems that `.modulerc` files are parsed twice on version 3.2 which leads to these `Duplicate version symbol` messages. The symbolic version is however correctly set, so I believe this message could be ignored on your test cases.

File: easybuild/easyblocks/generic/modulerc.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    def install_step(self):
+        """Do nothing."""
+        pass
+
+    def make_module_step(self, fake=False):
+        """Install .modulerc file."""
+        modfile_path = self.module_generator.get_module_filepath(fake=fake)
+        modulerc = os.path.join(os.path.dirname(modfile_path), '.modulerc')
+
+        if os.path.exists(modulerc) and not build_option('force'):
```
Comment (ocaisa) [inline easybuild/easyblocks/generic/modulerc.py]: This implies that you can only have one entry in the `.modulerc` per software package. Probably ok, but worth noting the limitation

File: easybuild/easyblocks/generic/modulerc.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            raise EasyBuildError("Found existing .modulerc at %s, not overwriting without --force", modulerc)
+
+        deps = self.cfg['dependencies']
+        if len(deps) != 1:
+            raise EasyBuildError("There should be exactly one dependency specified, found %d", len(deps))
+
+        # names should match
+        if self.name != deps[0]['name']:
```
Comment (ocaisa) [inline easybuild/easyblocks/generic/modulerc.py]: Given the discussion about have a `Pythoncore` (or similar) in https://github.com/easybuilders/easybuild-easyconfigs/pull/6537 I wonder if we should consider having exceptions here? In that case I could imagine using `.modulrerc` to alias back to the underlying installation (at least for `GCC`)

Comment (ocaisa) [reply easybuild/easyblocks/generic/modulerc.py]: Nah, forget that

File: easybuild/easyblocks/generic/modulerc.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            raise EasyBuildError("Name does not match dependency name: %s vs %s", self.name, deps[0]['name'])
+
+        # version to prefix of version to alias to
```
[the remark this answers is not in the document]
Comment (boegel) [reply easybuild/easyblocks/generic/modulerc.py]: Yes, exactly that, and you're right, the comment that was there didn't make any sense, it should have been something like `# version is prefix of version to alias to`, but yours is better :)

[the remark this answers is not in the document]
Comment (boegel) [reply easybuild/easyblocks/generic/modulerc.py]: Yes, exactly that, and you're right, the comment that was there didn't make any sense, it should have been something like `# version is prefix of version to alias to`, but yours is better :)

File: easybuild/easyblocks/generic/modulerc.py (hunk 1 above)
Comment (boegel) [reply easybuild/easyblocks/generic/modulerc.py]: Indeed, that's a rather strict limitation for now, which we can review later when the need for it pops up?

A whole bunch of questions come into play to relax this:

* What do we do with an existing `.modulerc` that was not created by EasyBuild?
* How do we allow both appending to an existing `.modulerc` and overwriting the existing one (from the `eb` command line, I don't think specifying this in the easyconfig makes much sense)?
* Should we detect potentially conflicting lines in an existing `.modulerc`?

Comment (boegel) [reply easybuild/easyblocks/generic/modulerc.py]: Indeed, that's a rather strict limitation for now, which we can review later when the need for it pops up?

A whole bunch of questions come into play to relax this:

* What do we do with an existing `.modulerc` that was not created by EasyBuild?
* How do we allow both appending to an existing `.modulerc` and overwriting the existing one (from the `eb` command line, I don't think specifying this in the easyconfig makes much sense)?
* Should we detect potentially conflicting lines in an existing `.modulerc`?

File: easybuild/easyblocks/generic/modulerc.py (hunk 2 above)
Comment (boegel) [reply easybuild/easyblocks/generic/modulerc.py]: Regardless of whether what popped into your head makes sense or not: starting out strict and relaxing it later is a lot easier than the other way around... :)

**Pull request merged**